passes/opt/wreduce.cc

```diff
@@ -342,17 +342,17 @@ struct WreduceWorker
 			}
 		}
 
-		if (cell->type.in("$pos", "$add", "$mul", "$and", "$or", "$xor"))
+		if (cell->type.in("$pos", "$add", "$mul", "$and", "$or", "$xor", "$sub"))
 		{
-			bool is_signed = cell->getParam("\\A_SIGNED").as_bool();
+			bool is_signed = cell->getParam("\\A_SIGNED").as_bool() || cell->type == "$sub";
```

Comment:
Without this, borrow extension doesn't occur correctly since: https://github.com/YosysHQ/yosys/blob/bfc7164af7bf64cb2fe5d00e87bbfead841a4dc2/passes/opt/wreduce.cc#L362

Otherwise proof fails with:
```
a = 4'd8;
b = 8'd0;
assign o = (b >> 4) - a;
```
Pre-wreduce: o = 9'd111111000
Post-wreduce (without `|| cell->type == "sub"`): o = 9'b000011000 where the 4 MSBs that were trimmed were fixed to zero instead of extending.